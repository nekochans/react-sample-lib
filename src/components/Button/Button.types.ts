import { MouseEventHandler } from 'react';

export type ButtonProps = {
  type: 'primary' | 'secondary' | 'error';
  text: string;
  onClick: MouseEventHandler<HTMLButtonElement>;
};
