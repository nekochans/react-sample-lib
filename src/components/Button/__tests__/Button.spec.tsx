/**
 * @jest-environment jsdom
 */
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import Button from '..';
import { ButtonProps } from '../Button.types';

test('Button', () => {
  const props: ButtonProps = {
    type: 'primary',
    text: 'Primary Button',
    onClick: () => {
      console.log('PrimaryButton Callback!');
    },
  };

  const { asFragment } = render(
    <Button
      type={props.type}
      text={props.text}
      onClick={props.onClick}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
