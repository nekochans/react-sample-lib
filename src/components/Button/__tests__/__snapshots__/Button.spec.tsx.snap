// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<DocumentFragment>
  <button
    class="button button-primary"
    type="button"
  >
    Primary Button
  </button>
</DocumentFragment>
`;
